src/mlagility/cli/report.py

```diff
@@ -1,521 +1,204 @@
-            [
-                "Model Name",
-                "Unsupported ops (Optimized ONNX)",
-            ]
-        )
+    # Create report dict
+    Path(report_dir).mkdir(parents=True, exist_ok=True)
+    report = {}
 
+    # Add results from all cache folders
+    for cache_dir in cache_dirs:
+
+        # List all yaml files available
         all_model_state_yamls = cache.get_all(path=cache_dir, file_type="state.yaml")
         all_model_state_yamls = sorted(all_model_state_yamls)
 
+        # Add each model to report
         for model_state_yaml in all_model_state_yamls:
+
+            # Models are identified my the build name
+            build_name = model_state_yaml.split("/")[-2]
+
+            # Load state
             state = build.load_state(state_path=model_state_yaml)
 
-            writer.writerow(
-                [
-                    state.config.build_name,  # Model Name
-                    _cleanOpList(
-                        state.info.opt_onnx_unsupported_ops
-                    ),  # Unsupported ops (Optimized ONNX)
-                ]
+            # Add model to report if it doesn't exist
+            if build_name not in report:
+                report[build_name] = BuildResults()
+
+            # Get model hash from build name
+            report[build_name].model_hash = state.config.build_name.split("_")[-1]
+
+            # Get model hash from build name
+            report[build_name].params = _update_numeric_attribute(
+                state.info.num_parameters, report[build_name].params
             )
 
+            # Extract labels (if any)
+            parsed_labels = labels.load_from_cache(cache_dir, build_name)
+            expected_attr_from_label = {
+                "name": "model_name",
+                "author": "author",
+                "class": "model_class",
+                "task": "task",
+            }
+            for label in expected_attr_from_label.keys():
+                results_attr = expected_attr_from_label[label]
+                if label not in parsed_labels.keys():
+                    report[build_name].__dict__[results_attr] = "-"
+                else:
+                    report[build_name].__dict__[results_attr] = parsed_labels[label][0]
+
+            # Get Groq latency and number of chips
+            groq_estimated_latency = get_estimated_groq_latency(build_name, cache_dir)
+            report[build_name].groq_estimated_latency = _update_numeric_attribute(
+                groq_estimated_latency, report[build_name].groq_estimated_latency
+            )
+            report[build_name].groq_chips_used = _update_numeric_attribute(
+                state.num_chips_used, report[build_name].groq_chips_used
+            )
+
+            # Reloading state after estimating latency
+            state = build.load_state(state_path=model_state_yaml)
+
+            # Get CPU latency
+            x86_output_dir = os.path.join(cache_dir, build_name, "x86_benchmark")
```

Comment:
For this, we should put the MeasuredPerformance into state.info as a dict and then read it back

OR

load a ORTModel object and use that to parse the perf

Reply:
omodel = ORTModel(state)
latency = omodel._mean_latency()
etc.

Probably want to rename `_mean_latency()` to `mean_latency()` as well

Reply:
Solved + fixed TRT/ORT bug with moving cache folders